### ISSUE TITLE
CMake with MSVC setup for debug mode

### ISSUE THREAD

**Issue by RooTender** [Auburn/FastNoise2#113]
I'm trying to add this library to my project by building it using CMake like this:
(*the `%%` variables are just batch script variables, the result is presented below*)
```cmake
cmake -DFASTNOISE2_TESTS=OFF -DFASTNOISE2_NOISETOOL=OFF -S .\lib\FastNoise2 -B .\lib\FastNoise2\build\%%i
cmake --build .\lib\%%i\build\%%j --config Debug --clean-first --parallel %jobs%
```
![image](https://github.com/Auburn/FastNoise2/assets/24989290/6a0c8661-3deb-4daa-818a-02ac582bed7b)

And in my `CMakeLists.txt` I've compile it:
```cmake
cmake_minimum_required (VERSION 3.13)

...

if (MSVC)
  if(CMAKE_BUILD_TYPE STREQUAL "Debug")
    add_compile_definitions(_ITERATOR_DEBUG_LEVEL=2)
    add_compile_definitions(_DEBUG=1)

    # Forcing static build
    add_compile_options(/MTd)
  endif()
endif()

...

# FastNoise2
find_library(fastnoise2_path NAMES "FastNoise*" PATHS "${CMAKE_SOURCE_DIR}/lib/FastNoise2/build/${ARCHITECTURE}/${CMAKE_BUILD_TYPE}")
target_link_libraries(${PROJECT_NAME} ${fastnoise2_path})
include_directories(PRIVATE "${CMAKE_SOURCE_DIR}/lib/FastNoise2/include")
```

But when I'm trying to build I'm getting errors:
```
E:\Projects\block-forge\out\build\x64-debug\FastNoiseD.lib(SmartNode.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MDd_DynamicDebug' doesn't match value 'MTd_StaticDebug' in human_interface_device.cpp.obj
E:\Projects\block-forge\out\build\x64-debug\FastNoiseD.lib(Metadata.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MDd_DynamicDebug' doesn't match value 'MTd_StaticDebug' in human_interface_device.cpp.obj
E:\Projects\block-forge\out\build\x64-debug\FastNoiseD.lib(FastSIMD.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MDd_DynamicDebug' doesn't match value 'MTd_StaticDebug' in human_interface_device.cpp.obj
E:\Projects\block-forge\out\build\x64-debug\FastNoiseD.lib(FastSIMD_Level_SSE2.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MDd_DynamicDebug' doesn't match value 'MTd_StaticDebug' in human_interface_device.cpp.obj
E:\Projects\block-forge\out\build\x64-debug\FastNoiseD.lib(FastSIMD_Level_SSE41.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MDd_DynamicDebug' doesn't match value 'MTd_StaticDebug' in human_interface_device.cpp.obj
E:\Projects\block-forge\out\build\x64-debug\FastNoiseD.lib(FastSIMD_Level_AVX2.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MDd_DynamicDebug' doesn't match value 'MTd_StaticDebug' in human_interface_device.cpp.obj
E:\Projects\block-forge\out\build\x64-debug\FastNoiseD.lib(FastSIMD_Level_AVX512.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MDd_DynamicDebug' doesn't match value 'MTd_StaticDebug' in human_interface_device.cpp.obj
```

I have defined the necessary macro in the header that directly links to that library:
![image](https://github.com/Auburn/FastNoise2/assets/24989290/26ea799f-c5bb-4aaa-95da-89e7c3a1e521)

Also, the release mode works perfectly (also static lib)

**Comment by Auburn** [comment]
If you want to use `add_compile_options(/MTd)` you'd need to also set it on the FastNoise project. Or remove it so they match

Although I don't think  `add_compile_options(/MTd)` is doing what you think it is, it defines how the code links to the C runtime. It doesn't affect how libraries like FastNoise are linked

**Comment by RooTender** [comment]
ALL RIGHT, fixed that 🎉. So it seems that I've needed to add this line:
```cmake
cmake_minimum_required (VERSION 3.15) # The var CMAKE_MSVC_RUNTIME_LIBRARY is available from the 3.15 version

...

if (MSVC)
  set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>DLL:Debug>") # Here it is
  if(CMAKE_BUILD_TYPE STREQUAL "Debug")
    add_compile_definitions(_ITERATOR_DEBUG_LEVEL=2)
    add_compile_definitions(_DEBUG=1)
  endif()
endif()
```

And actually what happened is that other libraries are linked statically **and** the MSVC RunTime is linked also. It's a dll file and therefore it required this configuration. Might be quite useful for someone who could be struggling with that.

PS: Thanks for quick response, appreciate that.
